blog/2020-05-16-web-support.md

```diff
@@ -0,0 +1,109 @@
+---
+title: React Navigation on the Web
+author: Satyajit Sahoo
+author_url: https://twitter.com/satya164
+author_title: Core Team
+author_image_url: https://avatars2.githubusercontent.com/u/1174278?s=200&v=4
+tags: [announcement, web]
+---
+
+React Native has made cross-platform development much easier than before, and with React Native for Web, you can reuse code across Android, iOS and Web too!
+
+One major pain point of reusing code for the web app has been navigation. React Navigation is one of the most widely used navigation libraries for React Native, but it didn’t support web. While you could run apps using React Navigation on the Web, a lot of things were missing, such as proper integration with URLs on the browser.
+
+We have finally added preliminary web support to React Navigation. Let's take a look at the changes.
+
+<!--truncate-->
+
+## Integration with URLs in browsers
+
+The first step for web support is to have proper URL integration. This means:
+
+- Users should be able to navigate to a screen in the app using a URL
+- URL in the browser's address bar should update as users navigates in the app
+- Users should be able to go backwards and forwards using browser's back/forward buttons
+- Buttons that navigate to other screens in the app should be links, and users should be able use standard shortcuts with them
+
+<video playsInline autoPlay muted loop style={{ maxWidth: '100%', marginBottom: 32 }}>
+  <source src="/assets/blog/web-support/url-integration.mp4" type="video/mp4" />
+</video>
+
+In native apps, it's already possible to navigate to a screen in the app using a URL via deep links. React Navigation can handle deep links if you provide a configuration in the `linking` prop of `NavigationContainer`. We now reuse the same configuration for URL integration in browsers. It also works for updating the URL in the browser when users navigates in the app. So if you have configured deep links in your app already, it'll work with URL integration on the web without any changes 🥳
```

Comment:
```suggestion
In native apps, it's already possible to navigate to a screen in the app using a URL via deep links. React Navigation can handle deep links if you provide a configuration in the `linking` prop of `NavigationContainer`. We now reuse the same configuration for URL integration in browsers. It also works for updating the URL in the browser when users navigate in the app. So if you have configured deep links in your app already, it'll work with URL integration on the web without any changes 🥳
```

Reply:
we might also want to mention that we will generate some default paths based on route names if no config is provided! i think that's a nice feature